v2/docs/OCPSTRAT-1515.md

```diff
@@ -0,0 +1,207 @@
+* **For OLM-v0 catalogs**, no perfect solution exists that ensures signature validity at the same time. Among the solutions mentioned:
+  * no filtering: which keeps the image intact, and inline with its signature. On the downside, clusters will still suffer from [OCPBUGS-35386](https://issues.redhat.com/browse/OCPBUGS-35386).
+  * a release concept attached to the catalog image, that can be used by OLM, the webconsole, etc to patch the original FBC. CatalogSource can have extra stanza, `patches` for example. But SQLite catalogs cannot be patched. Additionally, extra effort on a soon to be deprecated version is going to be a waste.
+  * mirroring by minVersion only: Contrary to v1, oc-mirror v2 would not allow filtering by maxVersion. As a result, more operator bundles will be mirrored, up to each channel head. The catalog image is also kept intact in this solution, so inline with its signature. The clusters won't experience [OCPBUGS-35386](https://issues.redhat.com/browse/OCPBUGS-35386) as the channel head bundle is always available. The downside of this solution is that oc-mirror customers might choose not to migrate to oc-mirror v2 because it lacks the filtering by maxVersion filtering.
+  * filtering by minVersion and maxVersion: Preserving the oc-mirror v1 functionality, and rebuilding the catalog image from the filtered FBC. The downside here is that the signature verification will fail, and that the SBOM will not correspond to the filtered catalog image. 
+
+## Selected solution for OLM-v0 catalogs
+
+This document will only focus on the feature design for OLM-v0 catalogs. Reflecting the mirrored operators in OLM-v1 catalogs shall be handled in a different feature request, at a later time.
+
+From the above solutions for OLM-v0 catalogs, the decision was made to rebuild catalog images inserting the FBC filtered by minVersion (and maxVersion) in the image. 
+
+**This solution is TEMPORARY.** It shall be accompanied by proper documentation explaining that filtered catalogs will not benefit from signature verification, and that this problem will be **fully solved by migrating to OLM-v1 catalogs.**
+
+## Regarding [OCPSTRAT-1417](https://issues.redhat.com/browse/OCPSTRAT-1417)
+
+For the switch from OLM v0 to OLM v1, oc-mirror would need to switch from rebuilding catalogs and generating CatalogSource objects (compatible with rebuilt catalogs, without signature verification) to creating catalog patches (filtering metadata) and ClusterCatalog objects (compatible with OLM-v1, supporting signature verification). In this manner, a user cannot by mistake use a rebuilt catalog on a cluster with OLM-v1, nor a OLM-v1 compatible catalog on a cluster with OLM-v0. 
+
+A signature policy needs to be generated by oc-mirror along with catalogSources in order to instruct the cluster that it should not verify signatures for OLM-v0 catalog images: Being rebuilt images, the catalog wouldn't be usable on the cluster if the signature verification is enforced.
```

Comment:
Given that `buildah/podman` is going to support signing images, that should help other Red Hat toolings (oc-mirror included) to build images and sign them for later signature verification. 

Since the mirrored and rebuilt catalogs are considered "custom catalogs", customers should be instructed to sign the rebuilt catalog images using their own private key.  

This way, they can still consume these rebuilt/custom catalogs in their clusters with the signature validation enforced (e.g., validating images from the official Red Hat registry or being in-house generated by themselves).